src/app/core/services/hall.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { ALL_CINEMAS_HALLS } from '../../../assets/data/halls';
+import { IHall } from '../../models/hall';
+import { map } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class HallService {
+
+  getHalls(): Observable<Array<IHall>> {
+    return of(ALL_CINEMAS_HALLS);
+  }
+
+  getHallById(cinemaId: number): Observable<IHall> {
+    return this.getHalls()
+      .pipe(map((halls: Array<IHall>) => halls.find(({ id }) => id === cinemaId) as IHall),
```

Comment:
As I remember we use a similar find function with movies. It would be better to extract this function to the seperate one, make it generic and reuse